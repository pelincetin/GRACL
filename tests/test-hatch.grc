int start_routine(NodeList nl){
    for (Node n in nl){
        print(n.data());
    }
    return 0;
}

int main(){
    Graph g;
    Node node1;
    Node node2;
    Node node3;
    Node node4;
    Node node5;
    Node node6;
    Node node7;
    Node node8;
    Node node9;
    NodeList nl1;
    g = createGraph(10);
    node1 = g.createNode("A");
    node2 = g.createNode("B");
    node3 = g.createNode("C");
    node4 = g.createNode("D");
    node5 = g.createNode("E");
    node6 = g.createNode("F");
    node7 = g.createNode("G");
    node8 = g.createNode("H");
    node9 = g.createNode("I");
    nl1 = g.nodes();

    hatch nl1 start_routine(nl1){
        print("Code executed by the parent thread");
    }
    return 0;
}