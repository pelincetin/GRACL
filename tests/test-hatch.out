Concurrency means non-deterministic output, this test is expected to fail